rfcs/008 - Best UX for long-running, blocking operations.md

```diff
@@ -0,0 +1,80 @@
+# 008 - Best UX for long-running, blocking operations like `iasql_install` and `iasql_commit` RFC
+
+### Proposed
+
+2023-03-20
+
+### Accepted
+
+YYYY-MM-DD
+
+#### Approvers
+
+- David Ellis <david@iasql.com>
+- Yolanda Robla <yolanda@iasql.com>
+- Alejandro Guillen <alejandro@iasql.com>
+- Mohammad Pabandi <mohammad@iasql.com>
+
+### Implementation
+
+- [ ] Implemented YYYY-MM-DD
+- [ ] Revoked/Superceded by: [RFC ###](./000 - RFC Template.md) YYYY-MM-DD
+
+## Author(s)
+
+- Luis Fernando De Pombo <l@iasql.com>
+
+## Summary
+
+IaSQL lets users connect an AWS account to a PostgreSQL DB and automatically backfill the database with existing cloud resources. This is meant to make the onboarding into IaSQL smooth as users don't need to redefine or reconcile existing infrastructure that was previously created via the AWS console or any sort of cloud management tooling. However, importing all existing cloud resources, even for just a single module, can take many minutes. This is exacerbated by users with nontrivial AWS accounts as install times seem to scale linearly with the number of resources involved. For example, if an S3 bucket or ECR repository has several hundred artifacts, it can take a long time to import the metadata for each bucket object or repository image. Reducing the install times is akin to squeezing blood from a stone since we run the risk of hitting the AWS SDK rate limiters on the other end. We need to find a better UX for module installation. Currently, the user waits for several minutes whilst no information is provided other than an animated loader. User feedback is one of the core pillars of good UX and we are failing at it for `iasql_install`, but also `iasql_commit`. Additionally, `iasql_install` is typically the first query that is run so users are not familiar with what a successful query is supposed to look like in the dashboard. We must also be mindful that not all users will go through the dashboard as they are welcome to use whatever PostgreSQL client they prefer. Finally, whatever UX we decide to go with should probably be mirrored in `iasql_uninstall` and `iasql_commit`. Particularly the latter which can also take a long time.
```

Comment:
The point about the users being allowed to use whatever postgres client they want could be a significant constraint on the options we can follow here.